### PR TITLE
sources/azure: refactor http_with_retries to remove **kwargs

### PR DESCRIPTION
**Pull request by cjp256**
Remove unused functionality and general kwargs support in favor
of defined parameters.

As only headers and data kwargs are used, add those explicitly
to http_with_retries().

There should be no behavioral changes.

Signed-off-by: Chris Patterson <cpatterson@microsoft.com>

### CODE REVIEW

**Review by anhvoms — APPROVED** [review]
LGTM

File: cloudinit/sources/helpers/azure.py [hunk 1]
```diff
@@ -330,34 +330,22 @@ def get_ip_from_lease_value(fallback_lease_value):
 
 
 @azure_ds_telemetry_reporter
-def http_with_retries(url, **kwargs) -> url_helper.UrlResponse:
+def http_with_retries(
+    url, *, headers: dict, data=None
```
Comment (blackboxsw) [inline cloudinit/sources/helpers/azure.py:334]: Since we are type hinting, let's get url too
```suggestion
    url: str, *, headers: dict, data=None
```

**Review by blackboxsw — COMMENTED** [review]
Just a minor nit and +1

File: cloudinit/sources/helpers/azure.py (hunk 1 above)
Comment (cjp256) [reply cloudinit/sources/helpers/azure.py]: done, for data too, thanks!

**Pull request merged**